core/internal/cltest/mocks.go

```diff
@@ -445,19 +445,30 @@ func (InstantClock) After(_ time.Duration) <-chan time.Time {
 // to resume computation on After.
 type TriggerClock struct {
 	triggers chan time.Time
+	t        testing.TB
 }
 
 // NewTriggerClock returns a new TriggerClock, that a test can manually fire
 // to continue processing in a Clock dependency.
-func NewTriggerClock() *TriggerClock {
+func NewTriggerClock(t testing.TB) *TriggerClock {
 	return &TriggerClock{
 		triggers: make(chan time.Time),
+		t:        t,
 	}
 }
 
 // Trigger sends a time to unblock the After call.
 func (t *TriggerClock) Trigger() {
-	t.triggers <- time.Now()
+	select {
+	case t.triggers <- time.Now():
+	case <-time.After(30 * time.Second):
+		t.t.Error("timeed out while trying to trigger clock")
+	}
```

Comment:
Was this a pain point? what about relying on the [-timeout](https://golang.org/cmd/go/#hdr-Testing_flags) test flag?

Reply:
Also see https://github.com/smartcontractkit/chainlink/pull/1771